lazy_router.go

```diff
@@ -108,8 +109,12 @@ func (r *LazyRouter) Dispatch(ctx context.Context, req Request) ResponseQueue {
 					}
 				}
 			}
+			// if there are no valid response from all routes
+			out <- NewErrorResponse(errors.ErrNoValidResponseFromRoutes(req.Protocol()))
+		} else {
+			// if no routes returned from strategy, return error
+			out <- NewErrorResponse(errors.ErrRouterStrategyReturnedEmptyRoutes(req.Protocol())).WithLabels(labels)
 		}
-		out <- NewErrorResponse(errors.ErrRouterStrategyReturnedEmptyRoutes(req.Protocol())).WithLabels(labels)
```

Comment:
this block is actually meant to be ran only when `if len(routes) == 0`, however when all routes fail, `ok` is set to false and the control flow return this error which can be misleading.

Reply:
Good catch!